src/api/app/controllers/webui/comments_controller.rb

```diff
@@ -1,20 +1,26 @@
 class Webui::CommentsController < Webui::WebuiController
   before_action :require_login
+  before_action :set_commentable_type, only: :create
   before_action :find_commentable, only: :create
 
   def create
-    comment = @commented.comments.new(permitted_params)
-    User.session!.comments << comment
-    @commentable = comment.commentable
+    if @commented.nil?
+      flash.now[:error] = "Failed to create comment: This #{@commentable_type.name.downcase} does not exist anymore."
+      render partial: 'layouts/webui/flash'
```

Comment:
This action is called from JS, no? So I don't think it's needed to render this.

Reply:
Unfortunately, it is needed, otherwise the request fails with no feedback. This way at least the user knows that the operation did not succeed.